src/tests/yamlFormat.test.ts

```diff
@@ -0,0 +1,31 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: MIT
+ */
+
+import * as assert from 'assert'
+import { FormattingOptions } from 'vscode-languageserver-types'
+import { getYamlLanguageService, Position, Range } from '../service'
+import { toDocument } from './utils/testUtilities'
+
+async function getFormat(yaml: string, range: Range, formattingOptions: FormattingOptions) {
+    const { textDoc } = toDocument(yaml, true)
+    const ls = getYamlLanguageService({})
+
+    return ls.format(textDoc, range, formattingOptions);
+}
+
+suite('ASL YAML format', () => {
+    test('Format does not remove comments', async () => {
+        const yaml = `
+# this is my comment
+hello: world # this is another comment
+`
+        const formattedText = await getFormat(yaml, Range.create(Position.create(0, 0), Position.create(yaml.length, yaml.length)), {
+            tabSize: 4,
+            insertSpaces: true
+        })
+
+        assert.equal(formattedText.length, 0, 'Expected input YAML and Formatted YAML to be the same')
```

Comment:
Just wondering, why is this comparison to 0? Should this not be comparing `formattedText.length` to `yaml.length`?

Reply:
Good point, didn't even see this. Looks like format range might be wrong.
The result definitely shouldn't be an empty string.

Reply:
I should probably reword the variable but formattedText is actually an array of TextEdits not a string. So in this case it's checking that no text edits need to be applied to the document after the text is formatted